calculator-app/src/main/java/camp/nextstep/edu/calculator/CalculatorActivity.kt

```diff
@@ -2,26 +2,35 @@ package camp.nextstep.edu.calculator
 
 import android.os.Bundle
 import android.widget.Toast
-import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import camp.nextstep.edu.calculator.data.db.RecordDatabase
+import camp.nextstep.edu.calculator.data.repository.RecordRepositoryImpl
 import camp.nextstep.edu.calculator.databinding.ActivityCalculatorBinding
+import camp.nextstep.edu.calculator.domain.RecordRepository
 
 class CalculatorActivity : AppCompatActivity() {
     private lateinit var binding: ActivityCalculatorBinding
 
-    private val viewModel: CalculatorViewModel by viewModels()
+    private lateinit var viewModel: CalculatorViewModel
 
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityCalculatorBinding.inflate(layoutInflater)
+
+        val database = RecordDatabase.getInstance(this)
+
+        val repository: RecordRepository = RecordRepositoryImpl(database)
```

Comment:
데이터베이스와 respository는 데이터모듈에 구현체가 존재할텐데요..!
궁극적으로는 뷰 역할을 하고 있는 app 모듈에서 이 구현체들을 직접 알면 모듈화가 잘 진행되지 않은 것입니다.
실제 구현체의 내용을 알고 있어야하는 부분은 어디인지 고민해보시고 아래와 같은 injector를 추가해보시면 좋을 것 같습니다 :)

```suggestion
    object Injector {
        fun provideRecordRepository(dao: 사용할 dao): RecordRepository {
            return RecordRepositoryImpl(context)
        }
    }
```

Reply:
`val repository: RecordRepository = Injector.provideRecordRepository(this)`
요렇게 수정하였는데요 
실제 구현체를 제공하기위해 구현체가 있는 부분인 data모듈에 Injector object가 위치하게되더라구용,

 presentation에서는 이를 호출하기 위해 presentation모듈에 data모듈의 의존성을 갖고 있는거로 이해되는데요 

hilt에서도 provider(의존성 제공자)가 실제 구현체의 내용있는 부분인 data 모듈에 자동으로 생성되어서 presenetation에서 data의존성을 갖는것도 동일한 이유인가요?

Reply:
Injector는 data 모듈에 위치하게 되는 것이 맞습니다!

> presentation에서는 이를 호출하기 위해 presentation모듈에 data모듈의 의존성을 갖고 있는거로 이해되는데요

이 부분에 대해 답변 드리면 먼저 presentation에서 이를 호출하기 위해 data 모듈의 의존성을 갖고 있는 것이 맞습니다.
다음 step의 스포를 살짝 하자면 data모듈에 있는 실제 구현체같은 것들을 presentation 모듈에서 직접 가져오지 못하게 하기 위해 internal 예약어를 이용하게 됩니다.
그렇게 진행하면 presentation 모듈에서 data 모듈의 의존성은 가지고 있지만 data 모듈의 실제 구현체들은 가져와서 사용할 수 없는 상황이 생기게 되거든요.

Reply:
> hilt에서도 provider(의존성 제공자)가 실제 구현체의 내용있는 부분인 data 모듈에 자동으로 생성되어서 presenetation에서 data의존성을 갖는것도 동일한 이유인가요?

거의 맞는데요~
hilt를 이용해도 provider의 역할은 data 모듈에 자동으로 생성되지 않고, 개발자가 직접 provider를 구현해주어야합니다. 물론 힐트에서 어노테이션을 이용하여 간단하게 처리할 수 있게 되어있어요..!

Reply:
상세한 설명 감사드립니다~